pkg/controller/backup_deletion_controller.go

```diff
@@ -291,38 +297,60 @@ func (c *backupDeletionController) processRequest(req *velerov1api.DeleteBackupR
 
 	backupStore, err := c.backupStoreGetter.Get(location, pluginManager, log)
 	if err != nil {
-		return errors.Wrap(err, "error getting the backup store")
+		errs = append(errs, err.Error())
 	}
 
-	actions, err := pluginManager.GetDeleteItemActions()
-	log.Debugf("%d actions before invoking actions", len(actions))
+	deleteItemActions, err := pluginManager.GetDeleteItemActions()
+	log.Debugf("%d deleteItemActions before invoking deleteItemActions", len(deleteItemActions))
 	if err != nil {
-		return errors.Wrap(err, "error getting delete item actions")
+		return errors.Wrap(err, "error getting delete item deleteItemActions")
+	}
+	deleteItemActionResolver := framework.NewDeleteItemActionResolver(deleteItemActions)
+
+	var itemSnapshots []*volume.ItemSnapshot
+	var itemSnapshotters []isv1.ItemSnapshotter
+
+	if features.IsEnabled(velerov1api.UploadProgressFeatureFlag) {
```

Comment:
InvokeDeleteActions downloads the tarfile and untars it to get all of the info and we need that to get the list of item snapshots.  When it exits it removes all of the untarred resources.  In order to split into two functions it would need tp either download twice or leave the info lying around.  I think it's easier to do the two actions together.